Add JUnit test code and other minor changes

debbiextan opened the pull request:
### Implement JUnit Testing
- [x] Help Command

- [x] Summary Command

- [x] Schedule Command

- [x] Tracker Command

### Others

- [x] Update manual.txt

- [ ] Implement interface Trackable for Task

- [x] Handle negative cases for Schedule & Summary Command

- [ ] Add support for Lessons in Summary and Schedule Command

debbiextan committed:
Add Summary, Schedule, Tracker and Help Command test code.
Update manual.txt
Make minor edits to Tracker isNotStarted method.

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

debbiextan committed:
Fix checkstyle errors.

kahlokkee commented on src/test/java/oof/command/ScheduleCommandTest.java:
```diff
@@ -0,0 +1,106 @@
+                schedule++;
+            }
+        }
+        assertEquals(schedule, scheduleTasks.getSize());
```
This assertion is always true because of lines 63-64. 

By comparing size of scheduleTask, you are not testing for "correctDescriptionWithoutTasks".

kahlokkee commented on src/test/java/oof/command/ScheduleCommandTest.java:
```diff
@@ -0,0 +1,106 @@
+    public void execute_correctDescriptionWithoutTasks_scheduleShown() throws CommandException, ParserException {
+        String date = "01-01-2020";
+        oof.executeCommand("schedule " + date);
+        TaskList scheduleTasks = new TaskList();
```
Do not misuse TaskList. Use an arraylist of Tasks if you want to track the number of tasks.

kahlokkee commented on src/test/java/oof/command/ScheduleCommandTest.java:
```diff
@@ -0,0 +1,106 @@
+                schedule++;
+            }
+        }
+        assertEquals(schedule, scheduleTasks.getSize());
```
Same issue as 62-67

kahlokkee commented on src/test/java/oof/command/SummaryCommandTest.java:
```diff
@@ -0,0 +1,116 @@
+        }
+        assertEquals(scheduleTasks.getSize(), 0);
+
+    }
```
How do you know that "tomorrow" has no tasks? The test will fail depending the day it is run on...

kahlokkee commented on src/test/java/oof/command/SummaryCommandTest.java:
```diff
@@ -0,0 +1,116 @@
+                schedule++;
+            }
+        }
+        assertEquals(schedule, scheduledTasks.getSize());
```
Same issue as Schedule Command 62-67

kahlokkee commented on src/test/java/oof/command/TrackerCommandTest.java:
```diff
@@ -0,0 +1,231 @@
+
+        oof.executeCommand("delete " + recent);
+    }
+}
```
Same issue as others. Your assertions and your expected scenario do not match.

kahlokkee commented on src/main/java/META-INF/MANIFEST.MF:
```diff
@@ -0,0 +1,3 @@
+Manifest-Version: 1.0
+Main-Class: oof.Oof
+
```
where did this come from

kahlokkee reviewed (changes requested):
Testing code is quite problematic as you are not testing/asserting the success scenario that your testing method says. And some of your testing methods are designed to never fail which defeats the purpose of doing unit testing.

jasperosy committed:
Merge branch 'master' into master

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

debbiextan committed:
Handle negative cases of Schedule and Summary Command.
Refactor scheduleByDate method in Schedule Command.
Add JUnit test for Delete Command.
Fix JUnit test for Schedule and Summary Command.

debbiextan committed:
Merge remote-tracking branch 'origin/master'

debbiextan committed:
Remove Summary Command Test.
Change Schedule Command Test date.

debbiextan committed:
Remove Summary Command Test.
Change Schedule Command Test date.
Remove MANIFEST file (from previous commit).

debbiextan committed:
Fix checkstyle errors.

debbiextan committed:
Fix checkstyle errors.
Refactor Delete Task Command Test execute correctIndex deleteTask method.

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

debbiextan committed:
Refactor TrackerCommandTest.
Implement Delete and List functions for Tracker Command.
Update manual.txt and UG with new Tracker Command syntax.

debbiextan committed:
Merge remote-tracking branch 'upstream/master'

# Conflicts:
#	src/main/java/oof/logic/command/organization/exceptions/ScheduleEmptyException.java
#	src/main/java/oof/logic/command/productivity/ScheduleCommand.java
#	src/main/java/oof/logic/command/productivity/SummaryCommand.java

debbiextan committed:
Refactor TrackerCommandTest.
Implement Delete and List functions for Tracker Command.
Update manual.txt and UG with new Tracker Command syntax.
Fix merge conflict.

Kenlhc committed:
Merge branch 'master' into master
